File: CyxbsMobile2019_iOS/NewQA/Model/GroupItem.m
//
//  GroupItem.m
//  CyxbsMobile2019_iOS
//
//  Created by 阿栋 on 2020/12/24.
//  Copyright © 2020 Redrock. All rights reserved.
//

#import "GroupItem.h"

@implementation GroupItem

MJCodingImplementation

- (instancetype)initWithDic:(NSDictionary *)dict {
    if ([super init]) {
        self.topic_id = dict[@"topic_id"];
        self.topic_logo = dict[@"topic_logo"];
        self.topic_name = dict[@"topic_name"];
        self.message_count = dict[@"new_mes_count"];
        self.is_follow = dict[@"is_follow"];
        self.introduction = dict[@"introduction"];
    }
    return self;
}

@end

File: CyxbsMobile2019_iOS/Discover/Discover_WeDate/GroupItem.swift
//
//  GroupItem.swift
//  CyxbsMobile2019_iOS
//
//  Created by coin on 2023/8/30.
//  Copyright © 2023 Redrock. All rights reserved.
//

import UIKit

struct GroupItem {
    
    /// 是否置顶
    var isTop: Bool 
    /// 组id
    var groupID: Int
    /// 组名
    var name: String
    /// 组包含人员
    var members: [StudentItem]
    
    init(dictionary dic: [String: Any]) {
        isTop = dic["is_top"] as! Bool
        groupID = dic["id"] as! Int
        name = dic["name"] as! String
        
        var array: [StudentItem] = []
        for item in dic["members"] as! [[String: Any]] {
            array.append(StudentItem(dictionary: item))
        }
        members = array
    }
}
